runtime/domain.c

```diff
@@ -1489,16 +1491,15 @@ void caml_interrupt_self(void) {
 
 void caml_reset_young_limit(caml_domain_state * dom_st)
 {
-  CAMLassert(dom_st == Caml_state);
   /* An interrupt might have been queued in the meanwhile; this
      achieves the proper synchronisation. */
   atomic_exchange(&dom_st->young_limit, (uintnat)dom_st->young_start);
-  if (caml_incoming_interrupts_queued()
+  dom_internal * d = &all_domains[dom_st->id];
+  if (atomic_load_relaxed(&d->interruptor.interrupt_pending)
```

Comment:
I think `caml_incoming_interrupts_queued` was an acquire load, which is replaced by a relaxed load here. Don't we need an acquire load to properly synchronize with the STW leader?